s3/sc-s3-synapse-ra.yaml

```diff
@@ -58,7 +58,13 @@ Resources:
             Sid: "ReadAccess"
             Effect: "Allow"
             Principal:
-              AWS: !Ref 'S3UserARNs'
+              !Join
+                - ""
+                  - "AWS "
+                  - !Join
+                    - ","
+                      - !Ref ${S3UserARNs}
+                      - "arn:aws:iam::325565585839:root"
```

Comment:
This looks odd. I don't think you need two joins. Don't need the `- "AWS"`. Does this template validate? I.e. if you run `aws cloudformation validate-template --template-body <path-to-your-file>`?

Reply:
To be more clear, it currently looks like this:
```
Principal:
    AWS: !Ref 'S3UserARNs'
```
and I think whatever you're going to do should look something like this:
```
Principal:
    AWS: !Join ...
```
Join is going to create a string. Whatever is on the right side of "AWS:" needs to be a list. So it's possible that after combining the root ARN with the S3UserARNs list you'll have to Fn::Split.

Reply:
maybe you can try this?
```
Principal: !Join
        - ','
        - 'arn:aws:iam::325565585839:root'
        - !Ref S3UserARNs
```

Reply:
The validation tool returns errors, but also errors with the file as it is in the master branch...

Reply:
This new commit validates OK